src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+        End Sub
+
+        <Fact>
+        Public Sub CombinePathTest_BaseDirectory_RelativePath()
+            Assert.Equal(FileSystem.CombinePath("C:\", "Test2"), IO.Path.Combine("C:\", "Test2"))
+        End Sub
+        <Fact>
+        Public Sub CopyDirectory_SourceDirectoryName_DestinationDirectoryName()
+            Dim TestBase As New FileIOTestBase
+            Dim FullPathToSourceDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "SourceDirectory")
+            Dim FullPathToTargetDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "TargetDirectory")
+            IO.Directory.CreateDirectory(FullPathToSourceDirectory)
+            For i As Integer = 0 To 5
+                CreateTestFile(TestBase, SourceData, IO.Path.Combine(FullPathToSourceDirectory, $"NewFile{i}"))
+            Next
+            FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory)
+            Assert.Equal(IO.Directory.GetFiles(FullPathToSourceDirectory).Count, IO.Directory.GetFiles(FullPathToTargetDirectory).Count)
+            For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
+                ' Ensure copy transferred written data
+                Using stream As New IO.StreamReader(IO.File.OpenRead(CurrentFile))
+                    Dim ReadData(SourceData.Length - 1) As Char
+                    stream.Read(ReadData, 0, SourceData.Length)
+                    Assert.Equal(SourceData, ReadData)
+                End Using
+            Next
+            IO.Directory.Delete(FullPathToTargetDirectory, recursive:=True)
+            IO.Directory.CreateDirectory(FullPathToTargetDirectory)
+            CreateTestFile(TestBase, SourceData, IO.Path.Combine(FullPathToTargetDirectory, $"NewFile0"))
+            Assert.Throws(Of IO.IOException)(Sub() FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub CopyDirectory_SourceDirectoryName_DestinationDirectoryName_OverwriteFalse()
+            Dim TestBase As New FileIOTestBase
+            Dim FullPathToSourceDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "SourceDirectory")
+            Dim FullPathToTargetDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "TargetDirectory")
+            IO.Directory.CreateDirectory(FullPathToSourceDirectory)
+            For i As Integer = 0 To 5
+                CreateTestFile(TestBase, SourceData, IO.Path.Combine(FullPathToSourceDirectory, $"NewFile{i}"))
+            Next
+            FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory, overwrite:=False)
+            Assert.Equal(IO.Directory.GetFiles(FullPathToSourceDirectory).Count, IO.Directory.GetFiles(FullPathToTargetDirectory).Count)
+            For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
```

Comment:
can move this code to a private function

Reply:
Do you mean the code below? I am not sure how to read PR Feedback comments, do they apply to the 4 lines shown or just the last line?
```
           For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
                ' Ensure copy transferred written data
                Using stream As New IO.StreamReader(IO.File.OpenRead(CurrentFile))
                    Dim ReadData(SourceData.Length - 1) As Char
                    stream.Read(ReadData, 0, SourceData.Length)
                    Assert.Equal(SourceData, ReadData)
                End Using
            Next
 ```
Then won't the assert happen in the private function instead of where the failure is?

Reply:
When a failure happens it shows the testname also. so we can debug which test has failed. so its safe to move following part to a function.
```
           For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
                ' Ensure copy transferred written data
                Using stream As New IO.StreamReader(IO.File.OpenRead(CurrentFile))
                    Dim ReadData(SourceData.Length - 1) As Char
                    stream.Read(ReadData, 0, SourceData.Length)
                    Assert.Equal(SourceData, ReadData)
                End Using
            Next
 ```

i will try to be more clear about moving stuff to functions